src/Remote/Importer.php

```diff
@@ -0,0 +1,162 @@
+     * } $source
+     */
+    public function importFunctionsFrom(string $scheme, string $package, ?string $file = null, ?string $version = null, ?string $vcs = null, ?array $source = null): void
+    {
+        if (!$this->allowsRemote()) {
+            throw new ImportError(sprintf('Remote imports are disabled, skipping import of "%s".', $package));
+        }
+
+        if (isset($this->imports[$package]) && ($requiredVersion = array_key_first($this->imports[$package])) !== $version) {
+            throw new ImportError(sprintf('The package "%s" is already required in version "%s", could not require it in version "%s"', $package, $requiredVersion, $version));
+        }
+
+        if (!preg_match('#^(?<organization>[^/]+)/(?<repository>[^/]+)$#', $package)) {
+            throw new InvalidImportFormat(sprintf('The import path must be formatted like this: "%s://<organization>/<repository>".', $scheme));
+        }
+
+        if ('composer' === $scheme) {
+            if (null !== $source) {
+                throw new InvalidImportFormat('The "source" argument is not supported for Composer/Packagist packages.');
+            }
+
+            $this->importPackageWithComposer($package, version: $version ?? '*', repositoryUrl: $vcs, file: $file);
+
+            return;
+        }
+
+        if ('package' === $scheme) {
+            if (null !== $version || null !== $vcs) {
+                throw new InvalidImportFormat('The "source" and "vcs" arguments are not supported for non-Composer packages.');
+            }
+            if (null === $source) {
+                throw new InvalidImportFormat('The "source" argument is required for non-Composer packages.');
+            }
+
+            $this->importPackageWithComposer($package, version: 'v1', source: $source, file: $file);
+
+            return;
+        }
+
+        throw new InvalidImportFormat(sprintf('The import scheme "%s" is not supported.', $scheme));
+    }
+
+    public function fetchPackages(): void
+    {
+        if (!$this->imports) {
+            $this->composer->remove();
+
+            return;
+        }
+
+        $input = GlobalHelper::getApplication()->getInput();
+
+        // Need to look for the raw options as the input is not yet parsed
+        $forceUpdate = null === $input->getParameterOption('--update-remotes', true);
+        $displayProgress = null !== $input->getParameterOption('--no-interaction', true);
+
+        $autoloadPath = PathHelper::getRoot() . Composer::VENDOR_DIR . 'autoload.php';
+
+        if (!file_exists($autoloadPath)) {
+            $forceUpdate = true;
+        }
+
+        $this->composer->update($forceUpdate, $displayProgress);
+
+        require_once $autoloadPath;
+
+        foreach ($this->imports as $package => $data) {
+            foreach ($data as $version => $files) {
+                foreach ($files as $file) {
+                    import(PathHelper::getRoot() . Composer::VENDOR_DIR . $package . '/' . ($file ?? ''));
```

Comment:
**Not for now**, But I think we should move the import function to a dedicated class
Functions should be shortcut only, without logic.
(wahoo, just check, and many other functions are bloated :/)

Reply:
Oopsie, I just did it :grin